### PR TITLE
Added Merge Semantics Code Snippet

### LINKED ISSUE #70682: MergeSemantics does not work with VoiceOver
Using `MergeSemantics` to combine `Text` and `Checkbox` doesn't work on iOS with VoiceOver.
On Android with Talkback this works as expected.

## Steps to Reproduce

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  bool _isSelected = false;
  void _valueChanged(bool value) => setState(() => _isSelected = value);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: MergeSemantics(
          child: Row(
            children: [
              Text('Item'),
              Checkbox(
                value: this._isSelected,
                onChanged: this._valueChanged,
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

```
[✓] Flutter (Channel stable, 1.22.3, on Mac OS X 10.15.7 19H15, locale en-CH)
    • Flutter version 1.22.3 at /Users/matthi/zhaw/p7/flutter
    • Framework revision 8874f21e79 (3 weeks ago), 2020-10-29 14:14:35 -0700
    • Engine revision a1440ca392
    • Dart version 2.10.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/matthi/Library/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_SDK_ROOT = /Users/matthi/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Xcode 12.3, Build version 12C5020f
    • CocoaPods version 1.10.0

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[!] IntelliJ IDEA Ultimate Edition (version 2019.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[!] IntelliJ IDEA Community Edition (version 2020.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[!] VS Code (version 1.42.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
```

1. Activate VoiceOver
2. Move focus to merged Text/Checkbox node
3. Double Tap to check/uncheck

### Expected Result
The state of the checkbox changes.

### Actual Result
Nothing changes, only the current state is read out.

### PR DESCRIPTION
**Pull request by rahul13199kumar**
## Description

Added a Code Snippet for 'MergeSemantics'

## Related Issues

 Try to Fixes #70682

## Tests

I added the following tests:
 N/A  because only docs are change .
## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x ] I signed the [CLA].
- [x ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x ] I updated/added relevant documentation (doc comments with `///`).
- [ x] All existing and new tests are passing.
- [x ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by YazeedAlKhalaf** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fflutter%2Fflutter%2Fpull%2F71970) for more info**.

@rahul13199kumar The google bot says you didn't sign the CLA so follow the guide in the reply of google bot to sign it and then mention the bot saying you signed it.

**Comment by rahul13199kumar** [comment]
CA003BAD-7B2E-4B63-B94C-F3231483B78F

 @googlebot I signed it!

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -7238,24 +7238,27 @@ class Semantics extends SingleChildRenderObjectWidget {
...
+///    children:{
+///     value: true,
+///     onChanged:(value){},
+///    },
+///    Text("Settings"),
+///    ],
+///  ),
+/// ),
+/// ...
+///
```
Comment (YazeedAlKhalaf) [inline packages/flutter/lib/src/widgets/basic.dart:7256]: The explanation you added is not clear and has grammatical errors

The code snippet you added is broken, either revert to the original or fix the errors.
Tip: run the code in a flutter project and when everything works just copy and paste into docs.

In your code you use `{}` but you should use `()` and in children `[]` and you removed the checkbox widget and kept only the properties

File: packages/flutter/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -7238,24 +7238,27 @@ class Semantics extends SingleChildRenderObjectWidget {
...
 /// {@end-tool}
 ///
+///
+///
+///
```
Comment (YazeedAlKhalaf) [inline packages/flutter/lib/src/widgets/basic.dart:7261]: extra spacing for what reason?

File: packages/flutter/lib/src/widgets/basic.dart [hunk 3]
```diff
@@ -7238,24 +7238,27 @@ class Semantics extends SingleChildRenderObjectWidget {
...
-///         onChanged: (bool value) => null,
-///       ),
-///       const Text("Settings"),
-///     ],
-///   ),
-/// )
-/// ```
+/// here is the sample code to use[MergeSemantics] to merge the semantics of  a [checkbox] and [Text] widget.
+///
+/// ...dart
```
Comment (YazeedAlKhalaf) [inline packages/flutter/lib/src/widgets/basic.dart:7244]: this should be three back ticks `
not
...

File: packages/flutter/lib/src/widgets/basic.dart [hunk 4]
```diff
@@ -7238,24 +7238,27 @@ class Semantics extends SingleChildRenderObjectWidget {
...
+///  child: ROW{
+///    children:{
+///     value: true,
+///     onChanged:(value){},
+///    },
+///    Text("Settings"),
+///    ],
+///  ),
+/// ),
+/// ...
```
Comment (YazeedAlKhalaf) [inline packages/flutter/lib/src/widgets/basic.dart:7255]: this should be three back ticks `
not
...

File: packages/flutter/lib/src/widgets/basic.dart [hunk 5]
```diff
@@ -7238,24 +7238,27 @@ class Semantics extends SingleChildRenderObjectWidget {
...
-///       Checkbox(
-///         value: true,
-///         onChanged: (bool value) => null,
-///       ),
-///       const Text("Settings"),
-///     ],
-///   ),
-/// )
-/// ```
+/// here is the sample code to use[MergeSemantics] to merge the semantics of  a [checkbox] and [Text] widget.
```
Comment (YazeedAlKhalaf) [inline packages/flutter/lib/src/widgets/basic.dart:7242]: I would keep it as is:
```
/// This snippet shows how to use [MergeSemantics] to merge the semantics of
/// a [Checkbox] and [Text] widget.
```

File: packages/flutter/lib/src/widgets/basic.dart [hunk 6]
```diff
@@ -7238,24 +7238,27 @@ class Semantics extends SingleChildRenderObjectWidget {
...
-///     children: <Widget>[
-///       Checkbox(
-///         value: true,
-///         onChanged: (bool value) => null,
-///       ),
-///       const Text("Settings"),
-///     ],
-///   ),
-/// )
-/// ```
```
Comment (YazeedAlKhalaf) [inline packages/flutter/lib/src/widgets/basic.dart:7256]: revert to this code and add your edits because the new code is not working

**Review by YazeedAlKhalaf — CHANGES_REQUESTED** [review]
Needs a lot of changes

File: packages/flutter/lib/src/widgets/basic.dart (hunk 1 above)
Comment (rahul13199kumar) [reply packages/flutter/lib/src/widgets/basic.dart:7256]: @YazeedAlKhalaf , Thanks for your suggestion. I am working on it .

**Comment by chunhtai** [comment]
Hi @rahul13199kumar , are you still working on this PR?

**Comment by rahul13199kumar** [comment]
@chunhtai  , Today I put PR for this issue.

Thankyou

**Comment by goderbauer** [comment]
@rahul13199kumar I am going to close this PR in favor of your other PR then.

**Pull request closed without merging**